### DISCUSSION TITLE
Need help debugging jobs

### DISCUSSION THREAD

**Discussion by gssj85** [spatie/laravel-multitenancy#530] [Q&A]
I have wrote this question many times but I think I can see the issue now so I will rewrite:

The default DB_CONNECTION set on .env should be the landlord or the tenant?

I have it set as the tenant and when a job starts it is using a tenant connection with empty host.

I think this may be the root cause since It may attempt to find the tenant in a "bogus" tenant connection.

**Comment by masterix21** [comment]
It would be best if you used the landlord connection as default.

**Reply by gssj85** [reply]
Thank you for answering, I'm going to try it, hopefully this is the issue.

Are you the maintainer?

I think that if a job has the property tenantId it should connect on the landlord connection by default, doesn't it make sense?

**Reply by masterix21** [reply]
> Are you the maintainer?

I'm a collaborator

**Comment by gssj85** [comment]
I managed to make it work doing a bit of manual work (using landlord as default connection as you recommended):

Job handle method:

```php
public function handle()
    {
        $id_tenant = $this->job->payload()['tenantId'];
        $tenant = Tenant::find($id_tenant);
        config([
            'database.connections.tenant.host"'=> $tenant->host,
            'database.connections.tenant.database' => trim($tenant->database),
            'database.connections.tenant.port"'=> $tenant->port ?? '3050',
            'database.connections.tenant.username' => $tenant->username,
            'database.connections.tenant.password' => $tenant->password,
        ]);
        config(['database.default' => 'tenant']);
        DB::reconnect('tenant');

        UsuarioDispositivo::find($this->id_usuario_dispositivo)->forceDelete();

        DB::purge('tenant');
    }
```
    
I intend to move this logic to Queue::before/after so all jobs have it.
    
That would fix my problem I think (it fixed direct in the job but that code is not final as it should apply to all jobs, that have a tenantId in the payload).

But I have a problem that I cannot pass a model to a job because it somehow keeps the job running.

I need help to debug where it is happening because if I pass anything like a int, string, etc instead of a model the job doesn't even fail or throw any error messages it just keeps "running" forever.

This works:

```php
public function __construct(int $id_usuario_dispositivo)
    {
        $this->id_usuario_dispositivo = $id_usuario_dispositivo;
    }
```

this doesn't work
```php
public function __construct(UsuarioDispositivo $usuario_dispositivo)
    {
         // etc..
    }
```

*It works once, job runs and done right after however if I send another one to the queue it is stuck.

I'm on Laravel 9, last version of the package, using Redis as queue driver.

I really want to find out what's happening...it sucks to not know where to go :/

I think is when something resolves the serializaed model but I lack the knowledge to go further, please advise!

**Reply by masterix21** [reply]
Please create a repo on GitHub with the issue and add me to it so I can test and determine what you are doing wrong.

**Reply by julesjanssen** [reply]
My guess: the `UsuarioDispositivo` object is a Model that uses the tenant connection.
On unserializing the job command (in the `MakeQueueTenantAwareAction` class), this tenant connection is unknown. This results in a query-exception & the job is simply skipped.
I don't know how to 'fix' this exactly, but I think it's simplest to pass a model ID instead of a model into the constructor & manually fetch the model instance in your `handle` method.

**Comment by gssj85** [comment]
Dont think I can share code from work :/

Give me your best guess 🙂

**Reply by masterix21** [reply]
I didn't mean to add myself to your work project but to create a new project in Laravel with a Job that has the same features that aren't working in your current project. Once you've done that and confirmed it doesn't work, add me, and I will help you.

**Reply by gssj85** [reply]
Project is way too complex to make a sample :/
